src/components/SearchResults/SearchResults.jsx

```diff
@@ -1,24 +1,48 @@
-// SearchResults.jsx
 import React, { useState } from "react";
 import TableHead from "./TableHead";
 import TableBody from "./TableBody";
 import CustomerProfile from "../CustomerProfile/CustomerProfile";
 
 const SearchResults = ({ bookings }) => {
   const [selectedProfileId, setSelectedProfile] = useState(null);
+  const [sortConfig, setSortConfig] = useState({
+    key: null,
+    direction: "ascending",
+  });
 
   function handleShowProfile(e) {
     setSelectedProfile(Number(e.target.value));
   }
+
+  function handleSort(key) {
+    let direction = "ascending";
+    if (sortConfig.key === key && sortConfig.direction === "ascending") {
+      direction = "descending";
+    }
+    setSortConfig({ key, direction });
+  }
+
+  const sortedBookings = [...bookings].sort((a, b) => {
+    if (sortConfig.direction === "ascending") {
+      return a[sortConfig.key] > b[sortConfig.key] ? 1 : -1;
+    } else {
+      return a[sortConfig.key] < b[sortConfig.key] ? 1 : -1;
+    }
```

Comment:
Well done Zeliha! very clear coding. I really loved it. One thing I would like to add is that you can modify sort function as case insensitive to ignore capital letter and convert every data to string before sorting.